File: 3227-find-missing-and-repeated-values/find-missing-and-repeated-values.py
class Solution:
    def findMissingAndRepeatedValues(self, grid: List[List[int]]) -> List[int]:
        n = len(grid)
        x = n*n
        freq = {}
        for row in grid:
            for num in row:
                freq[num] = freq.get(num, 0) + 1

        rep, miss = 0, 0 
        for i in range(1, x+1):
            if freq.get(i, 0) == 2:
                rep = i
            elif freq.get(i, 0) == 0:
                miss = i
        return [rep, miss]

File: 3227-find-missing-and-repeated-values/find-missing-and-repeated-values.cpp
class Solution {
public:
    vector<int> findMissingAndRepeatedValues(vector<vector<int>>& grid) {

        int n = grid.size();
        int total = n * n;

        vector<int> freq(total + 1, 0);

        for (auto &row : grid) {
            for (int num : row) {
                freq[num]++;
            }
        }

        int repeated = 0;
        int missing = 0;

        for (int i = 1; i <= total; i++) {
            if (freq[i] == 2)
                repeated = i;
            else if (freq[i] == 0)
                missing = i;
        }

        return {repeated, missing};
    }
};